cmd/clusterctl/client/cluster/topology_plan_test.go

```diff
@@ -0,0 +1,492 @@
+				if err != nil {
+					panic(err)
+				}
+				objects = append(objects, objs...)
+
+				return convertToPtrSlice(objects)
+			}(),
+			args: args{
+				in: func() *TopologyPlanInput {
+					objs, err := utilyaml.ToUnstructured(modifiedDockerMachineTemplateYAML)
+					if err != nil {
+						panic(err)
+					}
+					return &TopologyPlanInput{
+						Objs:              convertToPtrSlice(objs),
+						TargetClusterName: "my-cluster",
+					}
+				}(),
+			},
+			want: out{
+				affectedClusters: func() []client.ObjectKey {
+					cluster := client.ObjectKey{Namespace: "default", Name: "my-cluster"}
+					cluster2 := client.ObjectKey{Namespace: "default", Name: "my-second-cluster"}
+					return []client.ObjectKey{cluster, cluster2}
+				}(),
+				affectedClusterClasses: func() []client.ObjectKey {
+					cc := client.ObjectKey{Namespace: "default", Name: "my-cluster-class"}
+					return []client.ObjectKey{cc}
+				}(),
+				modified: []item{
+					{kind: "KubeadmControlPlane", namespace: "default", namePrefix: "my-cluster-"},
+				},
+				created: []item{
+					// Modifying the DockerClusterTemplate will result in template rotation. A new template will be created
+					// and used by KCP.
+					{kind: "DockerMachineTemplate", namespace: "default", namePrefix: "my-cluster-control-plane-"},
+				},
+				reconciledCluster: &client.ObjectKey{Namespace: "default", Name: "my-cluster"},
+			},
+			wantErr: false,
+		},
+		{
+			name:            "Input with objects in different namespaces should return error",
+			existingObjects: []*unstructured.Unstructured{},
+			args: args{
+				in: func() *TopologyPlanInput {
+					objs, err := utilyaml.ToUnstructured(objsInDifferentNamespacesYAML)
+					if err != nil {
+						panic(err)
+					}
+					return &TopologyPlanInput{
+						Objs: convertToPtrSlice(objs),
+					}
+				}(),
+			},
+			wantErr: true,
+		},
+		{
+			name: "Input with TargetNamespace different from objects in input should return error",
+			args: args{
+				in: func() *TopologyPlanInput {
+					objs, err := utilyaml.ToUnstructured(newClusterClassAndClusterYAML)
+					if err != nil {
+						panic(err)
+					}
+					return &TopologyPlanInput{
+						Objs:            convertToPtrSlice(objs),
+						TargetNamespace: "different-namespace",
+					}
+				}(),
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			existingObjects := []client.Object{}
+			for _, o := range tt.existingObjects {
+				existingObjects = append(existingObjects, o)
+			}
+			proxy := test.NewFakeProxy().WithClusterAvailable(true).WithFakeCAPISetup().WithObjs(existingObjects...)
+			inventoryClient := newInventoryClient(proxy, nil)
+			tc := newTopologyClient(
+				proxy,
+				inventoryClient,
+			)
+
+			res, err := tc.Plan(tt.args.in)
+			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
+				return
+			}
+			g.Expect(err).NotTo(HaveOccurred())
```

Comment:
```suggestion
			g.Expect(err).NotTo(HaveOccurred())

```
nit: can we add some new lines here and below just to make a bit easier to read / provide some structure